dbt/parser/docs.py

```diff
@@ -0,0 +1,93 @@
+import dbt.exceptions
+from dbt.node_types import NodeType
+from dbt.parser.base import BaseParser
+from dbt.contracts.graph.unparsed import UnparsedDocumentationFile
+from dbt.contracts.graph.parsed import ParsedDocumentation
+
+import jinja2.runtime
+import os
+
+
+class DocumentationParser(BaseParser):
+    @classmethod
+    def load_file(cls, package_name, root_dir, relative_dirs):
+        """Load and parse documentation in a lsit of projects. Returns a list
+        of ParsedNodes.
+        """
+        extension = "[!.#~]*.md"
+
+        file_matches = dbt.clients.system.find_matching(
+            root_dir,
+            relative_dirs,
+            extension)
+
+        for file_match in file_matches:
+            file_contents = dbt.clients.system.load_file_contents(
+                file_match.get('absolute_path'), strip=False)
+
+            parts = dbt.utils.split_path(file_match.get('relative_path', ''))
+            name, _ = os.path.splitext(parts[-1])
+
+            path = file_match.get('relative_path')
+            original_file_path = os.path.join(
+                file_match.get('searched_path'),
+                path)
+
+            yield UnparsedDocumentationFile(
+                root_path=root_dir,
+                resource_type=NodeType.Documentation,
+                path=path,
+                original_file_path=original_file_path,
+                package_name=package_name,
+                file_contents=file_contents
+            )
+
+    @classmethod
+    def parse(cls, all_projects, root_project_config, docfile):
+        try:
+            template = dbt.clients.jinja.get_template(docfile.file_contents,
+                                                      {})
+        except dbt.exceptions.CompilationException as e:
+            e.node = docfile
+            raise
+
+        profile = dbt.utils.get_profile_from_project(root_project_config)
+        schema = profile.get('schema', 'public')
+
+        for key, item in template.module.__dict__.items():
+            if type(item) != jinja2.runtime.Macro:
+                continue
+
+            if not key.startswith(dbt.utils.DOCS_PREFIX):
+                continue
+
+            name = key.replace(dbt.utils.DOCS_PREFIX, '')
+
+            # because docs are in their own graph namespace, node type doesn't
+            # need to be part of the unique ID.
+            unique_id = '{}.{}'.format(docfile.package_name, name)
```

Comment:
ah! I see. Is there any benefit to keeping the format of these unique ids consistent with the rest of dbt's unique ids?

Reply:
I don't think so. In the `nodes` and `macros` dicts, there are potentially multiple node types so the `.`-separated namespacing by type makes a lot of sense. In the `docs` dict (and, I hope, all future manifest members) there's only one type. I would have removed resource_type from the docs entirely if I could have. Docs can only be referenced via `doc()` and created via `{% docs ... %}`, and nothing else can be referenced that way, so you never have any ambiguity (like you might with seeds vs models and `ref`).

That said, if there are strong feelings about consistency, it's a pretty easy change.

Reply:
ok, I buy that